paasta_tools/cli/cmds/secret.py

```diff
@@ -337,7 +341,9 @@ def is_service_folder(soa_dir, service_name):
     return os.path.isfile(os.path.join(soa_dir, service_name, "service.yaml"))
 
 
-def _get_secret_provider_for_service(service_name, cluster_names=None, soa_dir=None):
+def _get_secret_provider_for_service(
+    service_name, cluster_names=None, soa_dir=None, secret_provider_extra_kwargs={}
+):
```

Comment:
this is technically fine right now, but it might be safer to refactor this to 
```suggestion
def _get_secret_provider_for_service(
    service_name, cluster_names=None, soa_dir=None, secret_provider_extra_kwargs=None
):
    secret_provider_extra_kwargs = secret_provider_extra_kwargs or {}
```
so that folks don't get surprised in the future by the behavior of mutable default parameters

Reply:
oh nice catch